uint16 FilterLength     # this is a number between 1 and 32. Default is 10
